src/pybind/mgr/volumes/fs/nfs.py

```diff
@@ -481,3 +482,28 @@ def delete_nfs_cluster(self, cluster_id):
         except Exception as e:
             log.warning("Failed to delete NFS Cluster")
             return -errno.EINVAL, "", str(e)
+
+    def show_nfs_cluster_info(self, cluster_id):
+        try:
+            self._set_cluster_id(cluster_id)
+            completion = self.mgr.list_daemons(daemon_type='nfs')
+            self.mgr._orchestrator_wait([completion])
+            orchestrator.raise_if_exception(completion)
+            # Here completion.result is a list DaemonDescription objects
+            for cluster in completion.result:
+                if self.cluster_id == cluster.service_id():
+                    """
+                    getaddrinfo sample output: [(<AddressFamily.AF_INET: 2>,
+                    <SocketKind.SOCK_STREAM: 1>, 6, 'xyz', ('172.217.166.98',2049)),
+                    (<AddressFamily.AF_INET6: 10>, <SocketKind.SOCK_STREAM: 1>, 6, '',
+                    ('2404:6800:4009:80d::200e', 2049, 0, 0))]
+                    """
+                    result = {'IP': [ip[4][0] for ip in socket.getaddrinfo(cluster.hostname, 2049,
+                                     flags=socket.AI_CANONNAME, type=socket.SOCK_STREAM)],
```

Comment:
add exception handling inside the loop so a single element does not break the list output?

e.g. `socket.gaierror: [Errno -2] Name or service not known`

Reply:
Done.